### PR TITLE
DCKUBE-435: Renamed the 'master' branch to 'main' and set it as default

### PR DESCRIPTION
**Pull request by nghazali**
Adjusted the change in documentation and references in repo

### CODE REVIEW

File: docs/docs/operations/resource_management/REQUESTS_AND_LIMITS.md [hunk 1]
```diff
@@ -6,10 +6,10 @@ The default resource requests that are used for each product are defined below.
 
 | Product  | CPU   |  Memory |
 |----------|:-----:|------:|
-| [Jira](https://bitbucket.org/atlassian-docker/docker-atlassian-jira/src/master/#markdown-header-memory-heap-size){.external}                        | `2`   | `2G`  |
-| [Confluence](https://bitbucket.org/atlassian-docker/docker-atlassian-confluence-server/src/master/#markdown-header-memory-heap-size){.external}     | `2`   | `2G`  |
-| [Bitbucket](https://bitbucket.org/atlassian-docker/docker-atlassian-bitbucket-server/src/master/){.external}                                        | `2`   | `2G`  |
-| [Crowd](https://bitbucket.org/atlassian-docker/docker-atlassian-crowd/src/master/){.external}                                                       | `2`   | `1G`  |
+| [Jira](https://bitbucket.org/atlassian-docker/docker-atlassian-jira/src/main/#markdown-header-memory-heap-size){.external}                        | `2`   | `2G`  |
```
Comment (nanux) [inline docs/docs/operations/resource_management/REQUESTS_AND_LIMITS.md:9]: These are still `master` in the Docker repositories. Please revert the change.

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ name: Maven unit tests
 on:
   push:
     branches:
-      - master
+      - main
```
Comment (nanux) [inline .github/workflows/maven.yml:9]: We can also remove `master` from here:
https://github.com/atlassian-labs/data-center-helm-charts/blob/main/.github/workflows/documentation.yaml#L5

File: src/main/charts/confluence/Changelog.md [hunk 3]
```diff
@@ -112,19 +112,19 @@ index a39d4ae..8e148f3 100644
 
 
 * Version 0.9.0 
-* Merge branch 'master' into DCKUBE-317-additional-libraries-for-synchron
```
Comment (errcode1202) [inline src/main/charts/confluence/Changelog.md:115]: Just wondering if we should be updating these changelogs? I can understand changelogs using `main` once this PR has been merged, but for historical reasons and provenance I think we should keep the existing changelogs as is.

File: src/main/charts/crowd/Changelog.md [hunk 4]
```diff
@@ -86,11 +86,11 @@ change.
 
 * Version 0.9.0 
 * initial commit - changed the max-body-size of request to 250MB for all products and documented in CONFIG.md (#140) 
-* Merge branch 'master' into dckube-267-define-minimum-compute-resources
```
Comment (errcode1202) [inline src/main/charts/crowd/Changelog.md:89]: Same as my previous comment on changelogs

File: src/main/charts/bitbucket/Changelog.md [hunk 5]
```diff
@@ -112,15 +112,15 @@ index d3342d2..786d343 100644
 * Version 0.9.0 
 * initial commit - changed the max-body-size of request to 250MB for all products and documented in CONFIG.md (#140) 
 * Update values.yaml (#139) 
-* Merge branch 'master' into dckube-267-define-minimum-compute-resources 
+* Merge branch 'main' into dckube-267-define-minimum-compute-resources
```
Comment (nanux) [inline src/main/charts/bitbucket/Changelog.md:115]: I am not sure if we should retrospectively change `Changelog.md`, the commits message will stay same.

**Review by errcode1202 — CHANGES_REQUESTED** [review]
Should the current/historical changelogs be updated? Ive added some comments on this, my thinking is modifying the changelogs is akin to modifying commit history. We shouldnt.

File: src/main/charts/bitbucket/templates/NOTES.txt [hunk 6]
```diff
@@ -52,4 +52,4 @@ Get the {{ title .Chart.Name }} URL by running these commands in the same shell:
 #################################################################################
 {{- end }}
 
-For further documentation, see https://github.com/atlassian-labs/data-center-helm-charts/tree/master/docs
\ No newline at end of file
+For further documentation, see https://github.com/atlassian-labs/data-center-helm-charts/tree/main/docs
```
Comment (nanux) [inline src/main/charts/bitbucket/templates/NOTES.txt:55]: Please revert the changes from all `NOTES.txt` files, I will update the files to point to our github pages instead.

File: src/main/charts/crowd/Changelog.md [hunk 7]
```diff
@@ -109,7 +109,7 @@ index a48f866..ce5a29d 100644
    resources:
      jvm:
 +      # -- JVM memory arguments below are based on the defaults defined for the Crowd docker container, see:
-+      # https://bitbucket.org/atlassian-docker/docker-atlassian-crowd/src/master/
++      # https://bitbucket.org/atlassian-docker/docker-atlassian-crowd/src/main/
```
Comment (nanux) [inline src/main/charts/crowd/Changelog.md:112]: This should be `master`

File: src/main/charts/crowd/values.yaml [hunk 8]
```diff
@@ -210,7 +210,7 @@ crowd:
 
     # JVM Memory / Heap Size definitions. These values below are based on the
     # defaults defined for the Crowd docker container.
-    # https://bitbucket.org/atlassian-docker/docker-atlassian-crowd-server/src/master/#markdown-header-memory-heap-size
+    # https://bitbucket.org/atlassian-docker/docker-atlassian-crowd-server/src/main/#markdown-header-memory-heap-size
```
Comment (nanux) [inline src/main/charts/crowd/values.yaml:213]: This should be `master`

**Review by nanux — CHANGES_REQUESTED** [review]

File: src/main/charts/bitbucket/Changelog.md (hunk 5 above)
Comment (errcode1202) [reply src/main/charts/bitbucket/Changelog.md:115]: Ive also made this point. Modifying historical changelogs is like modifying the commit history for a repo.

File: src/main/charts/bitbucket/templates/NOTES.txt (hunk 6 above)
Comment (nanux) [reply src/main/charts/bitbucket/templates/NOTES.txt:55]: This is the PR:
https://github.com/atlassian-labs/data-center-helm-charts/pull/233

File: src/main/charts/confluence/Changelog.md (hunk 3 above)
Comment (nghazali) [reply src/main/charts/confluence/Changelog.md:115]: Yes, however the master branch is not existed anymore and this is renamed to main - do you think we should still keep master?

**Review by errcode1202 — CHANGES_REQUESTED** [review]
Should we also be updating references to `master` in our documentation i.e: 
https://github.com/atlassian-labs/data-center-helm-charts/blob/main/docs/docs/examples/elasticsearch/BITBUCKET_ELASTICSEARCH.md

**Comment by nanux** [comment]
> Should we also be updating references to `master` in our documentation i.e:
> https://github.com/atlassian-labs/data-center-helm-charts/blob/main/docs/docs/examples/elasticsearch/BITBUCKET_ELASTICSEARCH.md
> 
> image

I think we should update our documentation but not guides that are referencing external entities. Our Elasticsearch fine-grained permission guide is following: https://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/es-createupdatedomains.html

You can search for master on the page. The same is true for the console itself:
Screen Shot 2021-08-10 at 10 37 21 am

When AWS will fix this, we should update our guide as well but if our guide is using different terms than the vendor UI, it will be confusing for users.

**Review by errcode1202 — APPROVED** [review]
🌯

File: src/main/charts/crowd/values.yaml (hunk 8 above)
Comment (nanux) [reply src/main/charts/crowd/values.yaml:213]: It is still `master`

File: docs/docs/README.md [hunk 9]
```diff
@@ -1,8 +1,8 @@
 # Atlassian Data Center Helm Charts
 
 [![Artifact Hub](https://img.shields.io/endpoint?url=https://artifacthub.io/badge/repository/atlassian-data-center)](https://artifacthub.io/packages/search?repo=atlassian-data-center)
-[![Atlassian license](https://img.shields.io/badge/license-Apache%202.0-blue.svg?style=flat-square)](https://github.com/atlassian-labs/data-center-helm-charts/blob/master/CONTRIBUTING.md) 
-[![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat-square)](https://github.com/atlassian-labs/data-center-helm-charts/blob/master/CONTRIBUTING.md) 
+[![Atlassian license](https://img.shields.io/badge/license-Apache%202.0-blue.svg?style=flat-square)](https://github.com/atlassian-labs/data-center-helm-charts/blob/main/CONTRIBUTING.md)
```
Comment (nanux) [inline docs/docs/README.md:4]: Can you also change `CONTRIBUTING.md` into `LICENSE`. I think this is just an oversight.

**Review by nanux — APPROVED** [review]
